### PR TITLE
Add "..." prefix to complementary error messages

### LINKED ISSUE #9317: [ide] Addressing the multiple errors problem
There a few things that keeps annoying me regarding IDE integration.
Sadly, for some of them there's nothing we can do expect wait for VSCode to give us enough control over the Problems.

One however we can do since we control the compiler is the following:

This kind of error:
image

Will turn into this single problem:

image

That's very annoying since you have to go back to the terminal view in order to get information about the actual type that was required or even actual error.

I suggest we deal with it by passing some env var to the compiler when run from VSHaxe, so it changes its output by grouping same-position error messages into a single message.

### PR DESCRIPTION
**Pull request by RealyUniqueName**
Closes #9317 

This PR modifies multi-line error messages by adding "..." prefix to complementary messages.

Before:
```
Main.hx:4: characters 28-31 : Class<Two> should be Class<Unknown<0> : One>
Main.hx:4: characters 28-31 : Constraint check failure for downcast.S
Main.hx:4: characters 28-31 : Two should be One
Main.hx:4: characters 28-31 : For function argument 'c'
```
After
```
Main.hx:4: characters 28-31 : Class<Two> should be Class<Unknown<0> : One>
Main.hx:4: characters 28-31 : ... Constraint check failure for downcast.S
Main.hx:4: characters 28-31 : ... Two should be One
Main.hx:4: characters 28-31 : ... For function argument 'c'
```

@Gama11 is that useful?

### CODE REVIEW

**Comment by Simn** [comment]
I can't answer if this is useful for IDEs, but I like the look of this regardless.

**Comment by Gama11** [comment]
Yeah, I think that should work for parsing error messages (once we actually do parse them).

File: src/compiler/haxe.ml [hunk 1]
```diff
@@ -1098,7 +1098,7 @@ with
 			ctx.messages <- [];
 		end else begin
 			error ctx (Printf.sprintf "You cannot access the %s package while %s (for %s)" pack (if pf = "macro" then "in a macro" else "targeting " ^ pf) (s_type_path m) ) p;
-			List.iter (error ctx "    referenced here") (List.rev pl);
+			List.iter (error ctx "... referenced here") (List.rev pl);
```
Comment (Gama11) [inline src/compiler/haxe.ml]: Shouldn't there be some helper function like `related_error` that handles the `... ` prefix so you can't get the formatting wrong (missing a space after or something) and it doesn't have to be repeated everywhere?

Comment (RealyUniqueName) [reply src/compiler/haxe.ml]: Done

**Comment by nadako** [comment]
This looks much nicer than the before as it's clear that it's a single error with some explanation, not at dozen \o/

**Comment by markknol** [comment]
Maybe a very `priority-netherlands` thing, but would it be nicer to use the official `…` ellipsis symbol, instead of 3 dots?

**Comment by Aurel300** [comment]
@markknol I don't think it would play very nice with non-Unicode terminals. (cough cough Windows)

**Comment by markknol** [comment]
Ok in that case nevermind 👍

**Pull request merged**